File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/model/MapPersonInformation.java
package de.limbusdev.guardianmonsters.fwmengine.world.model;

import com.badlogic.gdx.maps.MapObject;
import com.badlogic.gdx.maps.objects.RectangleMapObject;
import com.badlogic.gdx.math.MathUtils;

import de.limbusdev.guardianmonsters.geometry.IntVec2;

/**
 * Created by georg on 01.12.15.
 */
public class MapPersonInformation {
    /* ............................................................................ ATTRIBUTES .. */
    public String path;
    public IntVec2 startPosition;
    public boolean moves= false;
    public String name;
    public String conversation;
    public boolean male=false;
    public int spriteIndex=0;
    /* ........................................................................... CONSTRUCTOR .. */

    public MapPersonInformation(MapObject mo) {
        if(!mo.getProperties().containsKey("type") || !mo.getProperties().get("type").equals("person")) {
            throw new IllegalArgumentException("Given MapObject is not of type \"person\"");
        }

        construct(
            mo.getProperties().get("path", String.class),
            new IntVec2(
                MathUtils.round((((RectangleMapObject)mo).getRectangle().getX())),
                    MathUtils.round((((RectangleMapObject)mo).getRectangle().getY()))),
            Boolean.valueOf(mo.getProperties().get("static", String.class)),
            mo.getProperties().get("text", String.class),
            mo.getProperties().get("nameID", String.class),
            Boolean.parseBoolean(mo.getProperties().get("male", String.class)),
            Integer.parseInt(mo.getProperties().get("spriteIndex", String.class)));
    }

    public MapPersonInformation(String path, IntVec2 startPosition, boolean moves,
                                String conv, String name, boolean male, int spriteIndex) {
        construct(path, startPosition, moves, conv, name, male, spriteIndex);
    }

    private void construct(String path, IntVec2 startPosition, boolean moves,
                           String conv, String name, boolean male, int spriteIndex) {
        this.path = path;
        this.startPosition = startPosition;
        this.moves = moves;
        this.conversation = conv;
        this.male = male;
        this.spriteIndex = spriteIndex;
        if(name != null) this.name = name;
        else this.name = "";
    }
    /* ............................................................................... METHODS .. */
    
    /* ..................................................................... GETTERS & SETTERS .. */
}

File: GuardianMonsters/core/src/de/limbusdev/guardianmonsters/fwmengine/world/model/MapPersonInformation.kt
package de.limbusdev.guardianmonsters.fwmengine.world.model

import com.badlogic.gdx.maps.MapObject
import com.badlogic.gdx.maps.objects.RectangleMapObject
import com.badlogic.gdx.math.MathUtils
import de.limbusdev.guardianmonsters.fwmengine.world.ui.get

import de.limbusdev.utils.geometry.IntVec2

/**
 * Created by georg on 01.12.15.
 */
class MapPersonInformation
{
    // --------------------------------------------------------------------------------------------- PROPERTIES
    var path: String = ""
    var startPosition = IntVec2()
    var moves = false
    var name: String = ""
    var conversation: String = ""
    var male = false
    var spriteIndex = 0


    // --------------------------------------------------------------------------------------------- CONSTRUCTORS
    constructor(mo: MapObject)
    {
        check(mo.properties.containsKey("type") && mo.properties["type", "null"] == "person")
        { "Given MapObject is not of type \"person\"" }

        mo as RectangleMapObject

        construct(

                mo.properties["path", ""],
                IntVec2(MathUtils.round(mo.rectangle.x), MathUtils.round(mo.rectangle.y)),
                mo.properties["static", true],
                mo.properties["text", ""],
                mo.properties["personName", ""],
                mo.properties["male", true],
                mo.properties["spriteIndex", 0]
        )
    }

    constructor
    (
            path            : String,
            startPosition   : IntVec2,
            moves           : Boolean,
            conv            : String,
            name            : String,
            male            : Boolean,
            spriteIndex     : Int
    ) {
        construct(path, startPosition, moves, conv, name, male, spriteIndex)
    }

    fun construct
    (
            path            : String,
            startPosition   : IntVec2,
            moves           : Boolean,
            conv            : String,
            name            : String,
            male            : Boolean,
            spriteIndex     : Int
    ) {
        this.path = path
        this.startPosition = startPosition
        this.moves = moves
        this.conversation = conv
        this.male = male
        this.spriteIndex = spriteIndex
        this.name = name
    }
}
